src/cryptography/hazmat/backends/openssl/ec.py

```diff
@@ -182,6 +182,11 @@ def exchange(self, algorithm, peer_public_key):
                 _Reasons.UNSUPPORTED_EXCHANGE_ALGORITHM
             )
 
+        if type(peer_public_key.curve) is not type(self.curve):
```

Comment:
I know this was already fixed another way, but for future reference you could do it as a type check by saying:

```python
    if not isinstance(peer_public_key.curve, type(self.curve)):
```